python/mirascope/llm/formatting/resolve_format.py

```diff
@@ -0,0 +1,30 @@
+from . import _utils
+from .types import (
+    FormatInfo,
+    FormattingMode,
+    RequiredFormatT,
+)
+
+
+def resolve_format(
+    format: type[RequiredFormatT], *, default_mode: FormattingMode = "tool"
```

Comment:
the default default is "tool" because it works everywhere. Providers should choose their default explicitly, but for testing purposes etc it is helpful for this to have a default value for default_mode